LoopKit/WeakObserverSet.swift

```diff
@@ -8,41 +8,41 @@
 
 import Foundation
 
-public struct WeakObserverSetIterator<Observer>: IteratorProtocol {
-    let enumerator: NSEnumerator
-    
-    init(_ enumerator: NSEnumerator) {
-        self.enumerator = enumerator
-    }
-    
-    public mutating func next() -> Observer? {
-        return enumerator.nextObject() as? Observer
+public class WeakObserverSet<Observer>: Sequence {
+
+    public struct Iterator<Observer>: IteratorProtocol {
+        let enumerator: NSEnumerator
+
+        init(_ enumerator: NSEnumerator) {
+            self.enumerator = enumerator
+        }
+
+        public mutating func next() -> Observer? {
+            return enumerator.nextObject() as? Observer
+        }
     }
-}
 
-public class WeakObserverSet<Observer>: Sequence {
-    
     private var observers: NSHashTable<NSObject>
 
     public init() {
         self.observers = NSHashTable<NSObject>.weakObjects()
     }
-    
+
     public func add(_ observer: Observer) {
         guard let observerObject = observer as? NSObject else {
```

Comment:
If you'd like to soften this necessary runtime check from NSObject to any Swift class, it's possible to rewrite this type with some weakboxing. A gist of mine from when I last needed this a couple months ago: https://gist.github.com/mpangburn/d8b0d51b688fe6ffb10ef8d44dd15b85

It can't solve the 'protocols can't self-conform' problem, unfortunately.